autocertkit/network_tests.py

```diff
@@ -464,6 +464,15 @@ class VLANTestClass(testbase.NetworkTestClass):
     default_vlan = 800
     num_ips_required = 4
 
+    def __init__(self, session, config):
+        super(VLANTestClass, self).__init__(session=session, config=config)
+        
+        # CA-398896: Skip vlan test from NetworkTest
+        # by removing the network base_tag 'NA'
+        if self.config['vlantest'] is False:
+            if 'NA' in self.tags:
+                self.tags.remove('NA')
```

Comment:
- Should not expose base class privacy `NA` here
- `TestClass` already has `extra_init` to extend than `__init__`

So could be 
```
class TestClass(object):


    def mask():
        if self.base_tag in self.tags:
            self.tags.remove(self.base_tag)
            
            
class VLANTestClass(testbase.NetworkTestClass):


    def extra_init(self):
        if not self.config['vlantest']:
            self.mask()
        
        super(VLANTestClass, self).extra_init()
```

For long term `tags` and test generator etc need to be refactored as they are not useful, while this is unrelated to your current change.

Reply:
code updated.